src/ripple/rpc/handlers/LedgersHandler.cpp

```diff
@@ -0,0 +1,85 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Dev Null Productions, LLC
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/rpc/handlers/LedgersHandler.h>
+#include <ripple/app/ledger/LedgerToJson.h>
+#include <ripple/app/main/Application.h>
+#include <ripple/app/misc/LoadFeeTrack.h>
+#include <ripple/json/Object.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/jss.h>
+#include <ripple/resource/Fees.h>
+#include <ripple/rpc/impl/RPCHelpers.h>
+#include <ripple/rpc/Role.h>
+
+namespace ripple {
+namespace RPC {
+
+LedgersHandler::LedgersHandler (JsonContext& context) : context_ (context)
+{
+}
+
+Status LedgersHandler::check()
```

Comment:
A lot of code duplication between this function and `LedgerHandler::check()`. I think we should refactor to eliminate the code duplication